### ISSUE TITLE
Large compressed reports cannot be updated with PUT requests as documented

### ISSUE THREAD

**Issue by sfisher** [datacite/sashimi#135]
The page at https://support.datacite.org/docs/usage-reports-api-guide#sending-compressed-reports says that compressed reports can be updated in standard REST style by doing a PUT request with the ID on the end of the URL, however the service doesn't allow this and instead returns a "**401 unauthorized**" rather than allowing updates at a URL with the ID as documented.

It says: "`To update an existing compressed report make a PUT request`" and gives the example of "`PUT /reports/{report-id}`", which doesn't seem to work.

**Steps to duplicate**

1. Create a file called report.json with the contents shown at https://raw.githubusercontent.com/datacite/sashimi/master/spec/fixtures/files/report_compressed.json as used for testing this type of report.
2. Do the following steps to submit this report the first time at a bash shell.
```
# note the exact way that the zlib content body needs to be created for curl is very,
# very particular so you must create it this way.  Other attempts to use gzip switches
# or other zlib utilities to compress a file contents failed for me and I wasted many
# hours trying to get the exact right contents to make things work with curl.
cat report.json | gzip > report.gzipped

# set your api token
token="<your-token-goes-here>"

# use curl to submit this report as a compressed report using a POST request for initial submission
curl -i -H "Content-Type: application/gzip" -H "Content-Encoding: gzip" -H "Accept: gzip" -H "Authorization: Bearer $token" -X POST "https://api.test.datacite.org/reports" --data-binary "@report.gzipped" > output1.txt
```
3. Examine the output of the report in output1.txt and notice something like `{"report":{"id":"dace80f0-84fe-4460-88d4-7abc77828914" . . .` which should be used for subsequent PUT requests to update the report as explained in the documentation.

4. Now attempt to update this compressed report with a PUT request by adding the ID to the URL. (substitute below)
```
curl -i -H "Content-Type: application/gzip" -H "Content-Encoding: gzip" -H "Accept: gzip" -H "Authorization: Bearer $token" -X PUT "https://api.test.datacite.org/reports/<id-for-report>" --data-binary "@report.gzipped" > output2.txt
```

5. Examine the output of this PUT request to replace the report id with a new compressed report.  It shows `401 unauthorized status` as below.  The error message is also very deceptive and causes people to search for the wrong cause of the problem.

```
HTTP/2 401
date: Wed, 09 Dec 2020 23:39:29 GMT
content-type: application/json; charset=utf-8
status: 401 Unauthorized
cache-control: no-cache
vary: Accept-Encoding, Origin
x-request-id: bd16cc4b-0bfd-4f46-90b3-1a0d94c0edf8
x-runtime: 0.091749
x-powered-by: Phusion Passenger 6.0.6
server: nginx/1.18.0 + Phusion Passenger 6.0.6

{"errors":[{"status":"401","title":"You are not authorized to access this resource."}]}
```

It looks like maybe it updates the report if i leave the ID off the URL and use another POST or PUT request instead of trying to update from the report id.  This is weird, though, because usually PUTs take the ID of the item to be updated when using REST.

PS.  It also would be really helpful to document the way to create compressed file output with gzip and using curl since there are lots of ways in which it will not work correctly if you create a standard gzip file.  Also, I think perhaps the data-binary option may be required with curl rather than just `--data` or `-d` though I'm not 100% sure of this.

<img src="https://frontapp.com/assets/img/favicons/favicon-32x32.png" height="16" width="16" alt="Front logo" /> [Front conversations](https://app.frontapp.com/open/top_2oc5x)

**Comment by kjgarza** [comment]
I have managed to reproduce the 401 error but it seems related to the new JWT when updating a compressed report. There had not been  reports updates (in our database) since July for Dryad, which leads me to believe this is the first report update with the new credentials released in v0.10.2. Probably this is why we never saw this before. We will need to investigate further.

>It looks like maybe it updates the report if i leave the ID off the URL and use another POST or PUT request instead of trying to update from the report id. This is weird, though, because usually PUTs take the ID of the item to be updated when using REST.

If one were to remove the ID of the request on an existing report the API will either route you  to (A) add subset reports flow (see [doc](https://support.datacite.org/docs/usage-reports-api-guide#send-usage-report-in-subsets)) or (B) the UPSERT flow (see [PR](https://github.com/datacite/sashimi/issues/20)). 


We can definitely look at improving the docs to include manual report generation. Maybe we can manage it in a different ticket.

**Comment by svogt0511** [comment]
Fixed with pull request #13.
